libreyolo/label/page.py

```
@@ -0,0 +1,1824 @@
+    dup = (leak? `<div class="iwarn leak">${leak} duplicate group${leak>1?"s":""} leak across splits (train↔val) — this inflates your validation score.</div>`:"")
+      + `<div class="iwarn">${d.duplicate_groups.length} duplicate group${d.duplicate_groups.length>1?"s":""} · ${d.duplicate_image_count} images · Fix moves the extras to a reversible quarantine (keeps the train copy)</div>`
+      + d.duplicate_groups.slice(0,8).map(g=>
+        `<div class="idup">${g.ids.slice(0,7).map(id=>`<img class="ithumb" loading="lazy" src="/api/thumb/${id}">`).join("")}<span class="idsplit">${esc(g.splits.join(" + "))}</span><button class="ifix" data-ids='${JSON.stringify(g.ids)}'>Fix</button></div>`).join("");
+  } else {
+    dup = `<div class="iok">${ICO_CHECK}No duplicate or near-duplicate images detected</div>`;
+  }
+  $("#insbody").innerHTML =
+    readinessSection(lastStats, d, q)
+    + `<div class="igrid">`
+    + `<div class="icard"><div class="ik">Images</div><div class="iv">${d.measured}</div></div>`
+    + `<div class="icard"><div class="ik">Avg size</div><div class="iv">${W.mean}×${H.mean}</div></div>`
```

Ruff (0.15.15)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)